### ISSUE TITLE
[Bug]: WebKit crashes after 1662nd browser context has been created (and disposed)

### ISSUE THREAD

**Issue by LuohuaRain** [microsoft/playwright#8775]
### Playwright version

v1.14.1

### Operating system

Windows

### What browsers are you seeing the problem on?

WebKit

### Other information

microsoft/playwright-dotnet#1613 

[https://github.com/microsoft/playwright-dotnet/issues/1613#issuecomment-904364876](url)
> This is intended and not a bug. You need to limit the maximum parallel contexts which you have open. In your case you never close them. 

The truth is, I do close every context in every while loop.

> In other languages its called a worker group with a maximum concurrency limit.

I didn't use any concurrency/parallel tech.
As you can see, I used 
```C#
await using var context = await browser.NewContextAsync();
```
so browser **context count will awalys be 1**

My demo project is here: https://github.com/LuohuaRain/PlaywrightBug
I run it about 50 minutes and get exception `Protocol error (Dialog.enable): Browser closed.`
I really wonder why? It is really by design?

### What happened? / Describe the bug

![Snipaste_2021-08-28_22-07-34](https://user-images.githubusercontent.com/37979368/131220604-7c98c944-dfc4-4be7-8ed9-7666c50c3ef9.png)

1. [2021-08-28 21:17:58][1]response 200, browser context count:1
2. ......

n. [2021-08-28 22:04:51][1662]response 200, browser context count:1

### Code snippet to reproduce your bug

```C#
using Microsoft.Playwright;
using System;


using var playwright = await Playwright.CreateAsync();
await using var browser = await playwright.Webkit.LaunchAsync();
var i = 0;
while (true)
{
    try
    {
        i++;
        await using var context = await browser.NewContextAsync();
        var page = await context.NewPageAsync();
        var response = await page.GotoAsync("https://www.baidu.com");
        Console.WriteLine($"[{DateTime.Now:yyyy-MM-dd HH:mm:ss}][{i}]response {response.Status}, browser context count:{browser.Contexts.Count}");
    }
    catch (Exception ex)
    {
        Console.WriteLine(ex.Message);
        Console.ReadLine();
    }
}
```

### Relevant log output

```shell
[2021-08-28 22:04:47][1660]response 200, browser context count:1
[2021-08-28 22:04:49][1661]response 200, browser context count:1
[2021-08-28 22:04:51][1662]response 200, browser context count:1
Protocol error (Dialog.enable): Browser closed.
==================== Browser output: ====================
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777272) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777272) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777279) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777279) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777291) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777291) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777291) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777294) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777294) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777309) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777313) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777313) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777327) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777327) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777338) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777338) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777352) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777352) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777354) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777354) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777354) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777370) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777392) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777392) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777396) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777420) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777425) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777425) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777425) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777439) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777440) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777440) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777440) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777440) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777458) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777464) that does not exist.
[pid=5428][err] Attempted to create a NetworkLoad with a session (id=9223372036854777467) that does not exist.
[pid=5428][err] 1   00007FFB9B1B1209
[pid=5428][err] 2   00007FFB9B1B1089
[pid=5428][err] 3   00007FFB9B22B089
[pid=5428][err] 4   00007FFB9B1D5095
[pid=5428][err] 5   00007FFB9B227D1E
[pid=5428][err] 6   00007FFB9B1D8BFF
[pid=5428][err] 7   00007FFB9B22C389
[pid=5428][err] 8   00007FFBCB3A268A
[pid=5428][err] 9   00007FFBCC137974
[pid=5428][err] 10  00007FFBCEB1A2F1
[pid=5428] <process did exit: exitCode=3221225477, signal=null>
[pid=5428] starting temporary directories cleanup
[pid=5428] finished temporary directories cleanup
```

**Comment by avodovnik** [comment]
This repros on the node version as well, with a simple:

```js
const { webkit } = require('playwright');

(async () => {
  const browser = await webkit.launch();
  var i = 0;
  while(i++ < 2000) {
    const context = await browser.newContext();
    const page = await context.newPage();
    const response = await page.goto("https://www.microsoft.com");
    console.log(`${i}: ${response.status()} status`);
    await context.close();
  }

  await browser.close();
})();
```

it fails on 1662nd context/page.

**Comment by gigitalz** [comment]
https://github.com/microsoft/playwright/issues/6319#issuecomment-913012220

**Comment by mxschmitt** [comment]
Closing as part of the triage process since it seemed stale or did not get enough upvotes in the past. Please create a new issue with a detailed reproducible or feature request if you still face issues.

**Comment by mxschmitt** [comment]
Note: Should be fixed in v1.52 or [Canary](https://playwright.dev/docs/canary-releases).
